ui/src/criteria/attribute.tsx

```diff
@@ -81,52 +85,160 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+
+  const year = new Date().getFullYear();
+
+  if (
+    attributeFilterHint?.integerBoundsHint !== null &&
+    attributeFilterHint?.integerBoundsHint !== "undefined"
+  ) {
+    const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
+
+    if (
+      integerBoundsHint?.max !== null &&
+      typeof integerBoundsHint?.max !== "undefined" &&
+      integerBoundsHint?.min !== null &&
+      typeof integerBoundsHint?.min !== "undefined"
+    ) {
+      const min = Math.abs(year - integerBoundsHint.max);
+      const max = Math.abs(year - integerBoundsHint.min);
+
+      const [value, setValue] = React.useState<number[]>([min, max]);
+
+      const updateValue = (newValue: number[]) => {
+        setValue(newValue);
+        props.dispatchFn(
+          produce(props.data, (data) => {
+            data.range = newValue;
+          })
+        );
+      };
+
+      const handleChange = (event: Event, newValue: number | number[]) => {
+        updateValue(newValue as number[]);
+      };
+
+      const handleLeftInputChange = (
+        event: React.ChangeEvent<HTMLInputElement>
+      ) => {
+        const newValue =
+          event.target.value === "" ? min : Number(event.target.value);
+        updateValue([
+          Math.min(newValue, value[1]),
+          Math.max(newValue, value[1]),
+        ]);
+      };
+
+      const handleRightInputChange = (
+        event: React.ChangeEvent<HTMLInputElement>
+      ) => {
+        const newValue =
+          event.target.value === "" ? max : Number(event.target.value);
+        updateValue([
+          Math.min(value[0], newValue),
+          Math.max(value[0], newValue),
+        ]);
+      };
+
+      return (
+        <Box sx={{ width: 300 }}>
```

Comment:
@tjennison-google  I know we should not put the css there, but without the css, it will be as long as the screen.  I suppose we could not have another css file only for one box. Please advise. Thank you. Tim

Reply:
Can you specify it as a percentage instead (e.g. `width: "30%", minWidth: 100`)? That way it's still prevented from resizing too small and breaking but isn't being forced to a particular size.